appz_scripts/appz_models.py

```diff
@@ -1,42 +1,186 @@
-        self.name = paramsDict["name"]
-        self.isOptional = paramsDict["isOptional"]
+import re
+
+
+class Path:
+    def __init__(self, paths_dict):
+        self.app_path = paths_dict["app"]["path"]
+        self.app_query = paths_dict["app"]["query"]
+        self.web_path = paths_dict["web"]["path"]
+        self.web_query = paths_dict["web"]["query"]
+
+    def __str__(self):
+        return unicode(self)
+
+    def __unicode__(self):
+        return "app_path: {}\napp_query: {}\nweb_path: {}\nweb_query: {}"\
+            .format(
+                self.app_path,
+                self.app_query,
+                self.web_path,
+                self.web_query
+                )
+
+
+class Parameter:
+    def __init__(self, paramters_dict):
+        self.type = paramters_dict["type"]
+        self.name = paramters_dict["name"]
+        self.isOptional = paramters_dict["isOptional"]
+
+    def create_paramter_for_function(self):
+        optinal = "?" if self.isOptional else ""
+        return "{}: {}{}".format(self.name, self.type, optinal)
+
+    def __str__(self):
+        return unicode(self)
+
+    def __unicode__(self):
+        return "name: {}\ntype: {}\nOptional: {}".format(
+            self.name,
+            self.type,
+            self.isOptional
+            )
 
 
 class Action:
-    # TODO: finish actions
-    def __init__(self, actionDict):
-        self.name = actionDict["name"]
-        self.params = self.createParams(actionDict["params"])
-        self.paths = actionDict["paths"]
-
-    def createParams(self, params):
-        return [Param(param) for param in params]
-
-
-class App:
-    # jsonFile: is a dictionary created from json file
-    def __init__(self, jsonFile):
-        self.name = jsonFile["name"]
-        self.fallbackURL = jsonFile["fallbackURL"]
-        self.scheme = jsonFile["scheme"]
-        self.appStoreId = jsonFile["appStoreId"]
-        self.actions = self.createActions(jsonFile["actions"])
-
-    def createActions(self, actions):
+    def __init__(self, action_dict):
+        self.name = action_dict["name"]
+        self.parameters = self.create_params(action_dict["params"])
+        self.path = Path(action_dict["paths"])
+
+    def create_params(self, params):
+        return [Parameter(param) for param in params]
+
+    def get_parameter(self, parameter_string):
+        regex_query = re.findall('^{(.+?)}', parameter_string)
+        if len(regex_query) > 0:
+            return regex_query[0]
+        else:
+            return None
+
+    def create_function_definition(self):
+        definition = "case {}(".format(self.name)
+        for parameter in self.parameters:
+            definition += parameter.create_paramter_for_function()
+            definition += ","
+        definition = definition[:-1]  # remove tailing ',' character
+        definition += ")"
+        return definition
+
+    def create_action(self):
+        action = "case .{}(".format(self.name)
```

Comment:
@Dreamersoul Definitely starting to look a lot more organized. 

I also imagined that every part would be a template, and then we can have a template loader globally, `TemplateLoader.get("blah")`. Something like `parameter_enum_template.txt`, `parameter_arg_template.txt`, `action_enum_template.txt`, `action_definition_template.txt`, ... etc. This will make sure the script doesn't deal with any string manipulation, and trust Jinja for all that stuff.

This might be just a burden, so I leave it up to you whether it is worth it or not.